rclcpp/include/rclcpp/executor.hpp

```diff
@@ -107,11 +109,13 @@ class Executor
     }
   }
 
-  void spin_node_once(rclcpp::node::Node::SharedPtr & node, bool nonblocking = false)
+  template<typename Rep = int64_t, typename Period = std::milli>
+  void spin_node_once(rclcpp::node::Node::SharedPtr & node, std::chrono::duration<Rep,
+    Period> timeout = std::chrono::duration<Rep, Period>::zero())
```

Comment:
Does this change the default behavior? Before it would block by default, now it will send zero which I think should make it non-blocking? Maybe that's not right, but can I specify non-blocking and/or block forever and if so how?